File: Leet_Code-383/Leet_Code-383.py
'''
Question 383
'''

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        i = -1
        for c in ransomNote:
            i = magazine.find(c)
            if i == -1:
                return False
            else:
                magazine = magazine[:i] + magazine[i+1:]
        return True

File: Leet_Code-383/Leet_Code-383.java
/*
 * Question 383
 */

class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int i = -1;
        for (char c: ransomNote.toCharArray()){
            i = magazine.indexOf(c);
            if (i == -1){
                return false;
            } else {
                magazine = magazine.substring(0, i) + magazine.substring(i + 1);
            }
        }
        return true;
    }
}